pages/profile/application/edit.tsx

```diff
@@ -0,0 +1,646 @@
+  //       </h1>
+  //     );
+  //   }
+
+  // disable this for testing
+  if (!user) {
+    // If user haven't signed in, redirect them to login page
+    router.push('/auth');
+  }
+
+  if (loading) {
+    return <LoadIcon width={200} height={200} />;
+  }
+
+  //disables submitting form on enter key press
+  function onKeyDown(keyEvent) {
+    if ((keyEvent.charCode || keyEvent.keyCode) === 13) {
+      keyEvent.preventDefault();
+    }
+  }
+
+  const setErrors = (obj, values, errors) => {
+    if (obj.textInputQuestions)
+      for (let inputObj of obj.textInputQuestions) {
+        if (inputObj.required) {
+          if (!values[inputObj.name]) errors[inputObj.name] = 'Required';
+        }
+      }
+    if (obj.numberInputQuestions)
+      for (let inputObj of obj.numberInputQuestions) {
+        if (inputObj.required) {
+          if (isNaN(parseInt(values[inputObj.name]))) {
+            errors[inputObj.name] = 'Invalid number';
+          } else if (!values[inputObj.name] && values[inputObj.name] !== 0)
+            errors[inputObj.name] = 'Required';
+        }
+      }
+    if (obj.dropdownQuestions)
+      for (let inputObj of obj.dropdownQuestions) {
+        if (inputObj.required) {
+          if (!values[inputObj.name]) errors[inputObj.name] = 'Required';
+        }
+      }
+    if (obj.checkboxQuestions)
+      for (let inputObj of obj.checkboxQuestions) {
+        if (inputObj.required) {
+          if (!values[inputObj.name]) errors[inputObj.name] = 'Required';
+        }
+      }
+    if (obj.datalistQuestions)
+      for (let inputObj of obj.datalistQuestions) {
+        if (inputObj.required) {
+          if (!values[inputObj.name]) errors[inputObj.name] = 'Required';
+        }
+      }
+    if (obj.textAreaQuestions)
+      for (let inputObj of obj.textAreaQuestions) {
+        if (inputObj.required) {
+          if (!values[inputObj.name]) errors[inputObj.name] = 'Required';
+        }
+      }
+
+    return errors;
+  };
+
+  return (
+    <div className="flex flex-col flex-grow">
+      <Head>
+        <title>Hacker Application</title>
+        <meta name="description" content="Register for HackUTD 2024" />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <section className="pl-4 relative mb-4">
+        <Link href="/" passHref>
+          <div className="cursor-pointer items-center inline-flex text-white font-bold bg-[#40B7BA] rounded-lg px-3 py-2">
+            <ChevronLeftIcon className="text-white" fontSize={'large'} />
+            Home
+          </div>
+        </Link>
+      </section>
+
+      <section className="relative">
+        <Formik
+          initialValues={{
+            ...formInitialValues,
+            majorManual: '',
+            universityManual: '',
+            heardFromManual: '',
+            // have no idea why this works, but need to hard code it for the form values to overwrite the default
+            preferredEmail: user?.preferredEmail || '',
```

Comment:
I think `profile.user.preferredEmail` would work better in case registered email is not the same as login email.